frontend/src/metabase/visualizations/lib/settings/column.js

```diff
@@ -327,8 +327,7 @@ export const NUMBER_COLUMN_SETTINGS = {
     },
     default: "USD",
     getHidden: (column: Column, settings: ColumnSettings) =>
-      // NOTE: ideally we'd hide this if number_style != "currency" but that would result in a circular dependency
```

Comment:
This comment pointed to the potential issue. Since the `currency` setting now uses `number_style`, it should add that under `readDependencies` to ensure `number_style` is computed when we need it here. However, adding that would create a circular dependency between the two settings. I doubt we detect that situation well, so it might just infinite loop and crash the app.

In this case I think `getHidden` will only receive stale data if `number_style` is unset and we're computing a default. Given that, I guess leaving the dependency out is fine for native questions? The one situation I'd check is that for GUI questions with a default `number_style` we still display `currency`'s widget. If that's broken, this might need to be: `!isCurrency(column) && settings["number_style"] !== "currency"`

Reply:
Yup, it does, but won't take the default currency, which I believe it's another issue:
![Peek 2021-04-02 12-37](https://user-images.githubusercontent.com/1711649/113430307-4ff64980-93b0-11eb-9e48-6e677de1905c.gif)

What do you think? ready to merge?

Reply:
I agree not using the instance default is probably another issue. 

The situation I was curious about was whether the `currency` widget is still displayed when the default `number_style` is currency. In the gif, you set Count's `number_style` to currency, but Total should already have that as a default. If there are any setting dependency issues, I think that's where it would show up.

Reply:
Not picking up the defaults is #10787 (and 11394)